main/java/com/epam/app/HelloApp.java

```diff
@@ -0,0 +1,106 @@
+            String mode = scanner.nextLine();
+            Commands cmd = Commands.valueOf(mode);
+            switch (cmd) {
+                case TEST:
+                    runTests();
+                    break;
+                case RESET:
+                    Utils.resetUsers();
+                    break;
+                case DIAG:
+                    System.out.println("Please choose your user(User1, User2): ");
+                    String input = scanner.nextLine();
+                    String user = input;
+                    System.out.println("User chosen: " + input);
+                    System.out.println("Input exchange values");
+
+                    System.out.println("Input From curreny(USD, EUR, KZT, RUB): ");
+                    Currency fromCurrency = Currency.valueOf(scanner.nextLine());
+                    System.out.println("Input To curreny (USD, EUR, KZT, RUB): ");
+                    Currency toCurrency = Currency.valueOf(scanner.nextLine());
+                    System.out.println("Input currency ammount: ");
+                    BigDecimal amount = scanner.nextBigDecimal();
+                    scanner.nextLine();
+                    exchange(user, fromCurrency, toCurrency, amount);
+                    break;
+                case QUIT:
+                    exit = true;
+                    break;
+                default:
+                    System.out.println("Incorrect cmd.");
+            }
+            Thread.sleep(5000L);
+        }
+    }
+
+    public static void runTests() throws InterruptedException {
+        exchange("User1", Currency.USD, Currency.EUR, new BigDecimal(10));
+        exchange("User1", Currency.KZT, Currency.RUBLE, new BigDecimal(1000));
+
+        exchange("User2", Currency.EUR, Currency.USD, new BigDecimal(10));
+
+        exchange("User1", Currency.USD, Currency.EUR, new BigDecimal(20));
+        exchange("User1", Currency.KZT, Currency.RUBLE, new BigDecimal(500));
+
+        exchange("User2", Currency.RUBLE, Currency.KZT, new BigDecimal(2000));
+        exchange("User2", Currency.EUR, Currency.USD, new BigDecimal(50));
+        exchange("User2", Currency.RUBLE, Currency.KZT, new BigDecimal(1000));
+
+        exchange("User1", Currency.KZT, Currency.USD, new BigDecimal(3500));
+        exchange("User1", Currency.RUBLE, Currency.EUR, new BigDecimal(780));
+
+        exchange("User2", Currency.EUR, Currency.RUBLE, new BigDecimal(10));
+        exchange("User2", Currency.USD, Currency.KZT, new BigDecimal(10));
+
+        Thread.sleep(5000L);
+        if(Utils.checkUsers()){
+            System.out.println("Tests passed.");
+        }else{
+            System.out.println("Tests failed.");
+        }
+    }
+
+    public static void exchange(final String user, final Currency fromCurrency, final Currency toCurrency, final BigDecimal amount) {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
I am not sure that you can have any concurrency issues, cause you start your threads with time period.
Could you please user one ExecutorService(that works with multiply threads) to have more relistic picture)